### PR TITLE
Prepare some small changes for SGX attestation

### PR DESCRIPTION
**Pull request by lkatalin**
We will need an updated version of the SGX dependency and we'll need to use Intel's `aesm-proto` file to generate Rust structures during build.

### CODE REVIEW

File: src/protobuf/mod.rs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! This is a placeholder file for the Rust stuctures generated from
+//! aesm-proto.proto during build.
```
Comment (haraldh) [inline src/protobuf/mod.rs:4]:
```suggestion
//! Provides all generated protobuf modules

include!(concat!(env!("OUT_DIR"), "/protos/mod.rs"));
```

This then includes the generated protobuf modules.

You will then have to use `crate::protobuf::aesm_proto`.

**Review by haraldh — CHANGES_REQUESTED** [review]

File: build.rs [hunk 2]
```diff
@@ -111,6 +111,20 @@ fn main() {
...
+
+    Command::new("mv")
+        .current_dir(&Path::new(CRATE).join("src/protobuf"))
+        .arg("aesm_proto.rs")
+        .arg("mod.rs")
+        .status()
+        .unwrap_or_else(|_| panic!("Could not rename aesm proto file"));
+
     let out_dir = PathBuf::from(std::env::var("OUT_DIR").unwrap());
     let out_dir_bin = out_dir.join("bin");
```
[the remark this answers is not in the document]
Comment (lkatalin) [reply build.rs]: Thanks. I'm looking for info on `gen_mod_rs` but it's not very obvious. I assume it lets you name the output `mod.rs`, which is what I was trying to do.

**Comment by lkatalin** [comment]
I pulled the error handling for `create_dir()` out into its own function since we use it several times.

File: build.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (lkatalin) [reply build.rs]: Yeah, I see from the link this is how it's normally done: "And for now, you may generate that mod.rs in OUT_DIR file from build.rs, and include generated mod.rs from the src directory."

**Review by haraldh — APPROVED** [review]

**Review by connorkuehl — APPROVED** [review]
I did not confirm the correctness of Intel's proto file

**Pull request merged**